app/api/v2/identity/users.rb

```diff
@@ -16,12 +16,15 @@ class Users < Grape::API
         params do
           requires :email, type: String, desc: 'User Email', allow_blank: false
           requires :password, type: String, desc: 'User Password', allow_blank: false
+          optional :referral_id, type: String, desc: 'Referral uid'
           optional :captcha_response, types: [String, Hash],
                                       desc: 'Response from captcha widget'
         end
         post do
           declared_params = declared(params, include_missing: false)
           user_params = declared_params.slice('email', 'password')
+          user_params[:referral_id] = params[:referral_id] if params[:referral_id]
```

Comment:
we need to regex validate that it's a UID, I think they have a fixed size of 10 and stars with ID
ID[0-9]{8}